src/main/cpp/src/get_json_object.hpp

```diff
@@ -0,0 +1,1191 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "json_parser.hpp"
+
+#include <cudf/strings/string_view.hpp>
+#include <cudf/strings/strings_column_view.hpp>
+
+#include <thrust/optional.h>
+#include <thrust/pair.h>
+#include <thrust/scan.h>
+#include <thrust/tuple.h>
+
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace spark_rapids_jni {
+
+/**
+ * path instruction type
+ */
+enum class path_instruction_type { SUBSCRIPT, WILDCARD, KEY, INDEX, NAMED };
+
+namespace detail {
+
+/**
+ * write JSON style
+ */
+enum class write_style { raw_style, quoted_style, flatten_style };
```

Comment:
Nit: Enum class values should be `ALL_CAP`. In addition, they don't have to all have the suffix `_style`.
```suggestion
enum class write_style { RAW, QUOTED, FLATTEN };
```

Reply:
`src/main/cpp/src/JSONUtilsJni.cpp` refers to it.